@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -1302,6 +1309,79 @@ export class TextElementsRenderer {
             const layer = textCanvas.getLayer(textElement.renderOrder || DEFAULT_TEXT_CANVAS_LAYER);
 
             const isPathLabel = textElement.path !== undefined && !textElement.isLineMarker;
+            let screenPoints: THREE.Vector2[];
+            let indexOfFirstVisibleScreenPoint = -1;
+
+            if (isPathLabel) {
+                // Get the screen points that define the label's segments and create a path with
+                // them.
+                screenPoints = [];
+                let minX = Number.MAX_SAFE_INTEGER;
+                let maxX = Number.MIN_SAFE_INTEGER;
+                let minY = Number.MAX_SAFE_INTEGER;
+                let maxY = Number.MIN_SAFE_INTEGER;
+                const tileCenterX = textElement.tileCenterX!;
+                const tileCenterY = textElement.tileCenterY!;
+                for (const pt of textElement.path!) {
+                    tempPosition.set(pt.x + tileCenterX, pt.y + tileCenterY, 0);
+                    const screenPoint = this.m_screenProjector.project(tempPosition);
```

Comment:
<possibleprematureoptimization>
this is projecting to screen space, you could however be a slightly bit more performant and project to NDC and then instead of doing the screenBounds.contains, just check if the x / y point is outside -1<->+1
</possibleprematureoptimization>

Reply:
Added a function in ScreenProjector that does exactly that. But here, we need the screen points for later.

Reply:
You can always convert NDC to screen easily as well later, if you store the NDC points

Reply:
Keeping current code.